server/adaptors/integrations/repository/catalog_reader.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+type IntegrationPart = 'assets' | 'data' | 'schemas' | 'static';
+
+interface CatalogReader {
```

Comment:
the idea of an interface for catalog is great - but the method names for this interface are too explicit for file based - can u change the methods to be more generic so that both the `local_catalog_reader` and `network_catalog_reader` will have same common API
This interface should focus on the domain entities such as: 
 - getCatalogMetadata
 - getIntegrationMetadata
 - getIntegration

Reply:
I'm not sure this is what the catalog reader is for, because that's already what the `Repository` and `Integration` classes are doing. They do all of that abstraction of fetching and validating the parts of a catalog, the reader specifically is about the types of low-level scans (file or network) that they are able to do. `getCatalogMetadata` is already `Repository.getIntegrationList`, and `getIntegrationMetadata` is already `Integration.getConfig`.

Reply:
How do Repository and Integration handle network vs local ?
What is the network oriented corollary class to this one ?

Reply:
Have changed the names to make it better, and gave more responsibility to the class to remove some further file-specific handling from the calling code.